viewing_party/party.py

```diff
@@ -1,23 +1,154 @@
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+
+    return user_data
+
+
+def watch_movie(user_data, title):
+    new_watchlist = []
+    new_watched = []
+
+    for movie in user_data["watchlist"]:
+        new_watchlist.append(movie)
+    
+    for movie in user_data["watched"]:
+        new_watched.append(movie)
+
+    for movie in new_watchlist:
+        if movie["title"] == title:
+            new_watched.append(movie)  
+            new_watchlist.remove(movie)
+            break
+
+    updated_user_data = {
+        "watchlist": new_watchlist,
+        "watched": new_watched
+    }
+    
+    return updated_user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    if not len(user_data["watched"]):
+        return 0.0
+    sum = 0
+    for movie in user_data["watched"]:
+        sum += movie["rating"]
+
+    average = sum / len(user_data["watched"])
+    return average
+
+def get_most_watched_genre(user_data):
+    if not user_data:
+        return None
+
+    genres = {}
+    for movie in user_data["watched"]:
+        movie_genre = movie["genre"]
+        count = genres.get(movie_genre, 0)
+        genres[movie_genre] = count + 1
 
+    entry = None
+    max_number_of_occurences = 0
+
+    for genre, count in genres.items():
+        if count > max_number_of_occurences:
+            entry = genre
+            max_number_of_occurences = count
+
+    return entry
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    friend_movie = []
+
+    for friend_list in user_data["friends"]:
+        for movie in friend_list["watched"]:
+            if movie not in friend_movie:
+                friend_movie.append(movie)
+
+    unique_user_movie = []
+
+    for movie in user_data["watched"]:
+        if movie not in friend_movie:
+            unique_user_movie.append(movie)
+    return unique_user_movie
+
+
+def get_friends_unique_watched(user_data):
+    unique_user_movie = []
+
+    for movie in user_data["watched"]:
+        if movie not in unique_user_movie:
+            unique_user_movie.append(movie)
+
+    friend_unique_movie = []
+
+    for friend_list in user_data["friends"]:
+        for movie in friend_list["watched"]:
+            if movie not in unique_user_movie and movie not in friend_unique_movie:
+                friend_unique_movie.append(movie)
+
+    return friend_unique_movie
 
-        
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+
+def get_available_recs(user_data):
+    friend_unique_movie = get_friends_unique_watched(user_data)

```

Comment:
Nice job invoking a function you've already written so that you're not repeating code!!

Since `get_friends_unique_watched` returns a list of movies that several friends have watched, prefer the nouns in this variable to be plural like `friends_unique_movies`